### ISSUE TITLE
FileType.custom does not work with any file extension

### ISSUE THREAD

**Issue by luiscarielo83** [miguelpruivo/flutter_file_picker#798]
file_picker 4.0.0 doesn't work with 'json', 'dat', 'cw' (I invented it). 
I'm from my physical device (Galaxy S8, SDK 26). 
I need it to work with 'cw' extensions. The extension 'cw' I made it up. The file with the extension 'cw', is simply a text file.
Best regards.

**Comment by miguelpruivo** [comment]
Hi, this won't work and it's described on the FAQ at the Wiki. The extensions must be resolved to valid MIME types by the OS, custom types aren't supported because the OS doesn't recognise them as a valid MIME filters, hence you'll need to discard after picking by iterating through it. Same applies for some "known" extensions such as .csv which isn't supported for some reason.

TL;DR: It's an OS limitation.
